nomad/auth/auth.go

```diff
@@ -86,6 +86,10 @@ func NewAuthenticator(cfg *AuthenticatorConfig) *Authenticator {
 	}
 }
 
+// SetVerifyTLS is a helper method to set the verifyTLS field. This is used in
+// when the server TLS configuration is updated.
+func (s *Authenticator) SetVerifyTLS(verifyTLS bool) { s.verifyTLS = verifyTLS }
```

Comment:
I think we've got a data race here, right? The methods on this struct read the value concurrently. Would be nice if we can do this without introducing a lock everywhere though (i.e. we only have this one boolean value to update so could we do it with an atomic load/store? Or reconfigure and swap out the whole `Authenticator` atomically?)